docker-compose.yml

```diff
@@ -0,0 +1,19 @@
+version: '3'
+services:
+  fpm:
+    image: php:fpm
```

Comment:
For the uid/gid issue, well you can put a directive like `user: 1000:1000` here.

- https://docs.docker.com/compose/compose-file/#domainname-hostname-ipc-mac_address-privileged-read_only-shm_size-stdin_open-tty-user-working_dir
- https://github.com/docker-library/docs/blob/master/php/README.md#running-as-an-arbitrary-user

The thing is we won't be able to find a good default here. If you do `git clone` uid/gid will be changed to current host user's ID, which usually has no relation to the ID in the container system.

Usually the best way to handle this is to add an entrypoint script (e.g. [wordpress](https://github.com/docker-library/wordpress/blob/a4d4806224d98fb21d9549270f87e851190174c8/docker-entrypoint.sh#L49-L52)) before the container starts, which changes uids of all required files in the volume. However, I am not sure if this is what user will expect by default when they run docker-compose (they will lose ownership to those data files).

So eventually this will make this drop-in docker-compose.yml less useful. If there are examples of how this is being handled by other (PHP) projects who add docker-compose.yml in the project, it would be helpful.

Reply:
I'll leave it open for discussion. I don't have a personal preference here.

Reply:
Alright I think [wordpress](https://github.com/docker-library/wordpress/blob/a4d4806224d98fb21d9549270f87e851190174c8/docker-entrypoint.sh#L49-L52)'s entrypoint.sh is the one that does it right. Even if it's OK to change file ownership (in this docker-compose config it used a mount volume), I am not sure if adding another entrypoint script in DokuWiki is too much or not.

The other way is to add docs about these issues, but this really loses some of the out-of-the-box benefits of this file.